jetbrains-core/src/software/aws/toolkits/jetbrains/services/federation/psireferences/ArnPsiReferenceContributor.kt

```diff
@@ -17,9 +18,15 @@ class ArnPsiReferenceContributor : PsiReferenceContributor() {
             ) {
                 override fun accepts(o: Any?, context: ProcessingContext): Boolean {
                     if (o == null || o !is PsiElement) return false
+                    val manipulator = ElementManipulators.getManipulator(o)
+                    if (manipulator == null) return false
 
-                    if (o.text.contains("arn:")) {
-                        return true
+                    try {
+                        if (manipulator.getRangeInElement(o).substring(o.text).contains("arn:")) {
```

Comment:
Can we not just retrieve the `TextRange` directly off the `PsiElement`?

Reply:
the manipulators are language-aware and will strip out text that is not relevant to us (for example, stripping quotes in literal declarations, or only no-oping if we're not looking at a leaf element in the YAML case)